tests/drivers/can/api/src/canfd.c

```diff
@@ -238,6 +238,45 @@ ZTEST(canfd, test_canfd_get_capabilities)
 			  "CAN FD loopback mode not supported");
 }
 
+/**
+ * @brief Test sending CAN FD frame with too big payload.
+ */
+ZTEST(canfd, test_send_fd_dlc_out_of_range)
+{
+	struct can_frame frame = {
+		.flags = CAN_FRAME_FDF | CAN_FRAME_BRS,
+		.id = TEST_CAN_STD_ID_1,
+		.dlc = CANFD_MAX_DLC + 1U,
+	};
+	int err;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_RUNTIME_ERROR_CHECKS);
+
+	err = can_send(can_dev, &frame, TEST_SEND_TIMEOUT, NULL, NULL);
+	zassert_equal(err, -EINVAL, "wrong error on sending invalid frame (err %d)", err);
+}
+
+/**
+ * @brief Test error when CAN FD Error State Indicator (ESI) is send without FD format flag (FDF).
+ *
+ * CAN FD Error State Indicator (ESI) indicates that the transmitting node is
+ * in error-passive state. Only valid in combination with CAN_FRAME_FDF.
+ */
+ZTEST(canfd, test_send_fd_incorrect_esi)
```

Comment:
There is already negative test that checks error code returned by can_send() when CAN frame ID is invalid:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/can/api/src/classic.c#L799
Here, I add similar test. It checks that can_send() returns -EINVAL when CAN frame has invalid set of flags (explanation in lines 262-263, copied from API description).